### PR TITLE
SRX-VV7I2J Display environment information in Release Dialog

### PR DESCRIPTION
**Pull request by jaypontes**

### CODE REVIEW

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.scss [hunk 1]
```diff
@@ -2,60 +2,163 @@
...
+.env-card-app-locks {
+  display: flex;
+  position: relative;
+  top: 20px;
+  right: 36px;
+  flex-direction: row;
+  gap: 13px;
+}
+
+.env-card-buttons {
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.scss:138]: please put all these rules regarding subcomponents of release dialog, into `.release-dialog`, like this:
```
.release-dialog {
  .env-card-buttons {
...
```

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.scss [hunk 2]
```diff
@@ -2,60 +2,163 @@
 @use '@material/card/mdc-card';
 
 .release-dialog {
-    border-radius: 15px 15px 0px 0px;
-    background-color: transparent;
-    border-radius: 10px;
+  border-radius: 15px 15px 0px 0px;
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.scss:5]: please attach a screenshot to the PR of how it looks now.

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.scss [hunk 3]
```diff
@@ -2,60 +2,163 @@
...
+  margin-right: 36px;
+  margin-bottom: 17px;
+  display: flex;
+  flex-direction: row;
+  gap: 20px;
+  button {
+    padding: 8px 19px;
+  }
+}
+
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.scss:149]: overall there are a lot of css rules here, are these all necessary?
Please try to remove some.

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx [hunk 4]
```diff
@@ -14,24 +14,76 @@ You should have received a copy of the GNU General Public License
...
+            if (a.name === b.name) return 0;
+            return 1;
+        }
+        if (distance[a.name] < distance[b.name]) return -1;
+        return 1;
+    });
+    return sortedEnvs.reverse();
+};
+
+export const calculateDistanceToUpstream = (envs: Environment[]): EnvSortOrder => {
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:54]: please add a test for these functions

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx [hunk 5]
```diff
@@ -69,24 +121,100 @@ export const ReleaseDialog: React.FC<ReleaseDialogProps> = (props) => {
...
+                            }
+                        />
                     </div>
-                    <List className={classNames('release-env-list', className)}>
-                        {envs.map((env) => (
-                            <ListItem button key={env}>
-                                <ListItemText primary={env} />
-                            </ListItem>
+                    <ul className={classNames('release-env-list', className)}>
+                        {sortEnvironmentsByUpstream(envs).map((env) => (
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:145]: please extract some of this code into its own componet, e.g. `<EnvironmentList> or <Environment>`

**Review by sven-urbanski-freiheit-com — CHANGES_REQUESTED** [review]
.

File: services/frontend-service/src/ui/components/ReleaseCard/ReleaseCard.tsx [hunk 6]
```diff
@@ -64,7 +64,11 @@ export const ReleaseCard: React.FC<ReleaseCardProps> = (props) => {
                 </div>
                 <div className="release__environments">
                     {environments.map((env) => (
-                        <Chip className={'release-environment release-environment--' + env} label={env} key={env} />
+                        <Chip
+                            className={'release-environment release-environment--' + env}
```
[the remark this answers is not in the document]
Comment (jaypontes) [reply services/frontend-service/src/ui/components/ReleaseCard/ReleaseCard.tsx]: done!

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.scss (hunk 2 above)
Comment (jaypontes) [reply services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.scss:5]: Screenshot 2022-12-19 at 15 45 16

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx (hunk 5 above)
Comment (jaypontes) [reply services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx]: Created EnvironmentList component!

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.scss (hunk 1 above)
Comment (jaypontes) [reply services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.scss]: Done!

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**